### ISSUE TITLE
CTC decoding script

### ISSUE THREAD

**Issue by joyolee** [facebookresearch/av_hubert#34]
Dear authors,

thanks a lot for this great work! After reading your paper I would ask 2 questions about the CTC decoding script.
In your paper B.4 you have also conducted experiments with CTC finetuning. In the current repo I didn't find the code for CTC finetuning. I would ask if it's also possible to share your CTC finetuning script?
The second question is that have you also done the experiments with  **cross entropy loss + CTC loss**? as it's widely used in acoustic speech recognition, it is also the default setting in the ESPNet toolkit. 

Best regards,
Zhengyang Li

**Comment by chevalierNoir** [comment]
Hi,

Thanks for your interest.

1. In the current repo, we haven't included a recipe for CTC decoding mainly because it performs worse than seq2seq and also requires several additional preparation steps including lexicion/language model preparation and installation of some libraries related to decoding. We plan to provide an example for CTC decoding in the future. But for now, we do have a model class for CTC fine-tuning [here](https://github.com/facebookresearch/av_hubert/blob/main/avhubert/hubert_asr.py#L153). The overall decoding process for CTC is similar to [HuBERT](https://github.com/pytorch/fairseq/tree/afc77bdf4bb51453ce76f1572ef2ee6ddcda8eeb/examples/hubert#decode-a-hubert-model).
2. By **cross-entropy loss+CTC loss**, did you mean joint training with seq2seq cross-entropy loss and CTC loss? We haven't tried it in our experiments but it probably can bring some additional gains. There is one prior work doing this: [Ma et al. 2021, End-to-end audio-visual speech recognition with conformers](https://arxiv.org/pdf/2102.06657.pdf).
